command/fmt_test.go

```diff
@@ -0,0 +1,52 @@
+package command
+
+import (
+	"path/filepath"
+	"strings"
+	"testing"
+
+	packersdk "github.com/hashicorp/packer/packer-plugin-sdk/packer"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFmt(t *testing.T) {
+	s := &strings.Builder{}
+	ui := &packersdk.BasicUi{
+		Writer: s,
+	}
+	c := &FormatCommand{
+		Meta: testMeta(t),
+	}
+
+	c.Ui = ui
+
+	args := []string{"-check=true", filepath.Join(testFixture("fmt"), "formatted.pkr.hcl")}
+	if code := c.Run(args); code != 0 {
+		fatalCommand(t, c.Meta)
+	}
+	expected := ""
+	assert.Equal(t, expected, strings.TrimSpace(s.String()))
```

Comment:
Totally for another PR but one nice thing we could have done is having a `input` file and an `expected` per folder, making the tests easy to read & add.